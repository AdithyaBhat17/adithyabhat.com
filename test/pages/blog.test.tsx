import { render } from '../testUtils'
import { cleanup } from '@testing-library/react'
import Blog from '@/pages/blog'
import Article from '@/pages/blog/[slug]'
import { mockArticles, mockArticle } from '../__mocks__/article'

afterAll(cleanup)

test('Renders a list of blogs', () => {
  const { container } = render(
    <Blog
      data={{ allArticles: mockArticles.allArticles }}
      type="allArticles"
      columns="3"
    />
  )
  expect(container).toMatchSnapshot()
})

test('Renders an article', () => {
  const { getByTestId } = render(
    <Article data={{ article: mockArticle, moreArticles: [] }} />
  )
  expect(getByTestId('title').textContent).toEqual('Intro to JavaScript')
  expect(getByTestId('date').textContent).toEqual('20 Aug, 2020')
  expect(getByTestId('content').textContent).toEqual('Blog contents')
})

test('Displays error page for invalid slug', () => {
  const { getByText } = render(
    <Article
      data={{ article: { ...mockArticle, slug: '' }, moreArticles: [] }}
    />
  )
  expect(getByText('404')).toBeInTheDocument()
})
