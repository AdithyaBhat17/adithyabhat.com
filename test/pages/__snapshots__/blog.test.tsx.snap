// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a list of blogs 1`] = `
<div>
  <div>
    <nav
      class="navbar px-6 md:px-36 lg:px-48"
    >
      <div
        class="flex justify-start items-center"
      >
        <a
          class="hover:border-white"
          href="/"
        >
          <img
            alt="Adithya NR logo"
            class="fill-current lg:h-16 lg:w-16 w-12 h-12"
            src="/static/logo_black.svg"
          />
        </a>
        <span
          class="sm:inline hidden ml-6 text-gray-500 capitalize"
        />
      </div>
      <ul
        class="flex justify-between"
      >
        <li
          class="ml-6"
        >
          <a
            class=" text-black text-sm hover:border-gray-900"
            data-testid="nav-link"
            href="/about"
          >
            About
          </a>
        </li>
        <li
          class="ml-6"
        >
          <a
            class=" text-black text-sm hover:border-gray-900"
            data-testid="nav-link"
            href="/blog"
          >
            Blog
          </a>
        </li>
        <li
          class="ml-6"
        >
          <a
            class=" text-black text-sm hover:border-gray-900"
            data-testid="nav-link"
            href="/work"
          >
            Work
          </a>
        </li>
        <li
          class="ml-6"
        >
          <a
            class=" text-black text-sm hover:border-gray-900"
            data-testid="nav-link"
            href="/contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="px-6 md:px-36 lg:px-48 mb-6"
      data-testid="container"
    >
      <h1
        class="font-semibold text-5xl text-gray-900 mb-6"
        style="opacity: 0; transform: translateY(60px) translateZ(0);"
      >
        Articles
      </h1>
      <div
        class="flex items-start flex-wrap -mx-5"
      >
        <div
          class="w-full md:w-1/3 mt-8 cursor-pointer text-gray-900 px-5"
          style="opacity: 0; transform: translateY(20px) translateZ(0);"
        >
          <div
            class="rounded-lg shadow-sm"
            style="display: block; overflow: hidden; position: relative;"
          >
            <img
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSI1NjkiIGhlaWdodD0iNDI3Ij48L3N2Zz4="
              style="display: block; width: 100%;"
            />
            <div
              style="background-image: url(data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAoHBwgHBgoICAgLFBILDhUNGRUSDh0PEQ8NFx8aGBYVKhUaLisjIh0oHSEiJDUlKC0vMjIyGSU4PTcwPCsxMi8BCgsLDg0OHBAQHC8oGh07Oy8vLy8vLy8vLy8vOzAvLy8vLy8vLy8vLy8vLy8vLy8vLy8vLy8vLy8vLy8vLy8vL//AABEIABIAGAMBIgACEQEDEQH/xAAaAAACAgMAAAAAAAAAAAAAAAAABAEDAgUG/8QAIRAAAgEEAAcAAAAAAAAAAAAAAAECAwQFERIjMTM0UXH/xAAXAQEBAQEAAAAAAAAAAAAAAAADBAUC/8QAGREAAgMBAAAAAAAAAAAAAAAAAAECETES/9oADAMBAAIRAxEAPwC6U1oQuKLqjFrz1syutUI7YkOU7EEaVtwMClZOnKfCmiStNCWbHDdtEZ3x5fAAy4YcPThbOUnkaicn19kABasFZ//Z); background-size: cover; opacity: 1; transition: opacity 500ms 500ms; position: absolute; left: 0px; top: 0px; bottom: 0px; right: 0px;"
            />
            <noscript />
          </div>
          <h2
            class="mt-3 font-semibold text-xl sm:text-sm lg:text-xl"
          >
            A beginner’s guide to understanding React’s context API
          </h2>
          <p
            class="text-gray-600 poppins sm:text-xs md:text-md mt-2"
          >
            Sep 2018 | React.js
          </p>
        </div>
        <div
          class="w-full md:w-1/3 mt-8 cursor-pointer text-gray-900 px-5"
          style="opacity: 0; transform: translateY(20px) translateZ(0);"
        >
          <div
            class="rounded-lg shadow-sm"
            style="display: block; overflow: hidden; position: relative;"
          >
            <img
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSI1NjkiIGhlaWdodD0iNDI3Ij48L3N2Zz4="
              style="display: block; width: 100%;"
            />
            <div
              style="background-image: url(data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAoHBwgHBgoICAgLFBILDhUNGRUSDh0PEQ8NFx8aGBYVKhUaLisjIh0oHSEiJDUlKC0vMjIyGSU4PTcwPCsxMi8BCgsLDg0OHBAQHC8oGh07Oy8vLy8vLy8vLy8vOzAvLy8vLy8vLy8vLy8vLy8vLy8vLy8vLy8vLy8vLy8vLy8vL//AABEIABIAGAMBIgACEQEDEQH/xAAaAAACAgMAAAAAAAAAAAAAAAAABAEDAgUG/8QAIRAAAgEEAAcAAAAAAAAAAAAAAAECAwQFERIjMTM0UXH/xAAXAQEBAQEAAAAAAAAAAAAAAAADBAUC/8QAGREAAgMBAAAAAAAAAAAAAAAAAAECETES/9oADAMBAAIRAxEAPwC6U1oQuKLqjFrz1syutUI7YkOU7EEaVtwMClZOnKfCmiStNCWbHDdtEZ3x5fAAy4YcPThbOUnkaicn19kABasFZ//Z); background-size: cover; opacity: 1; transition: opacity 500ms 500ms; position: absolute; left: 0px; top: 0px; bottom: 0px; right: 0px;"
            />
            <noscript />
          </div>
          <h2
            class="mt-3 font-semibold text-xl sm:text-sm lg:text-xl"
          >
            The most underrated skill for a developer: Communication
          </h2>
          <p
            class="text-gray-600 poppins sm:text-xs md:text-md mt-2"
          >
            Aug 2018 | Miscellaneous 
          </p>
        </div>
        <div
          class="w-full md:w-1/3 mt-8 cursor-pointer text-gray-900 px-5"
          style="opacity: 0; transform: translateY(20px) translateZ(0);"
        >
          <div
            class="rounded-lg shadow-sm"
            style="display: block; overflow: hidden; position: relative;"
          >
            <img
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSI1NjkiIGhlaWdodD0iNDI3Ij48L3N2Zz4="
              style="display: block; width: 100%;"
            />
            <div
              style="background-image: url(data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAoHBwgHBgoICAgLFBILDhUNGRUSDh0PEQ8NFx8aGBYVKhUaLisjIh0oHSEiJDUlKC0vMjIyGSU4PTcwPCsxMi8BCgsLDg0OHBAQHC8oGh07Oy8vLy8vLy8vLy8vOzAvLy8vLy8vLy8vLy8vLy8vLy8vLy8vLy8vLy8vLy8vLy8vL//AABEIABIAGAMBIgACEQEDEQH/xAAaAAACAgMAAAAAAAAAAAAAAAAABAEDAgUG/8QAIRAAAgEEAAcAAAAAAAAAAAAAAAECAwQFERIjMTM0UXH/xAAXAQEBAQEAAAAAAAAAAAAAAAADBAUC/8QAGREAAgMBAAAAAAAAAAAAAAAAAAECETES/9oADAMBAAIRAxEAPwC6U1oQuKLqjFrz1syutUI7YkOU7EEaVtwMClZOnKfCmiStNCWbHDdtEZ3x5fAAy4YcPThbOUnkaicn19kABasFZ//Z); background-size: cover; opacity: 1; transition: opacity 500ms 500ms; position: absolute; left: 0px; top: 0px; bottom: 0px; right: 0px;"
            />
            <noscript />
          </div>
          <h2
            class="mt-3 font-semibold text-xl sm:text-sm lg:text-xl"
          >
            Generate code coverage reports for your unit tests using Jest 
          </h2>
          <p
            class="text-gray-600 poppins sm:text-xs md:text-md mt-2"
          >
            Aug 2020 | Unit testing
          </p>
        </div>
      </div>
    </div>
    <br />
    <div
      class="px-6 md:px-36 lg:px-48 mb-6"
      data-testid="container"
    >
      <hr
        class="border-gray-800 mt-12 px-6"
      />
      <br />
      <footer
        class="footer flex-wrap"
      >
        <div
          class="flex justify-start items-center"
        >
          <a
            class="hover:border-white flex justify-start items-center"
            href="/"
          >
            <img
              alt="Adithya NR logo"
              class="fill-current lg:h-16 lg:w-12 w-12 h-12 mr-3"
              src="/static/logo_black.svg"
            />
            <span
              class="font-semibold text-black poppins"
            >
              adithya nr
            </span>
          </a>
        </div>
        <ul
          class="flex justify-between"
        >
          <li
            class="ml-0 md:ml-10"
          >
            <a
              class=" text-black text-sm hover:border-gray-900"
              data-testid="footer-link"
              href="/about"
            >
              About
            </a>
          </li>
          <li
            class="ml-0 md:ml-10"
          >
            <a
              class=" text-black text-sm hover:border-gray-900"
              data-testid="footer-link"
              href="/blog"
            >
              Blog
            </a>
          </li>
          <li
            class="ml-0 md:ml-10"
          >
            <a
              class=" text-black text-sm hover:border-gray-900"
              data-testid="footer-link"
              href="/work"
            >
              Work
            </a>
          </li>
          <li
            class="ml-0 md:ml-10"
          >
            <a
              class=" text-black text-sm hover:border-gray-900"
              data-testid="footer-link"
              href="/contact"
            >
              Contact
            </a>
          </li>
          <li
            class="ml-0 md:ml-10"
          >
            <a
              class=" text-black text-sm hover:border-gray-900"
              data-testid="footer-link"
              href="/resume.pdf"
            >
              Resume
            </a>
          </li>
          <li
            class="ml-0 md:ml-10"
          >
            <a
              class=" text-black text-sm hover:border-gray-900"
              data-testid="footer-link"
              href="/uses"
            >
              Uses
            </a>
          </li>
          <li
            class="ml-0 md:ml-10"
          >
            <a
              class=" text-black text-sm hover:border-gray-900"
              data-testid="footer-link"
              href="/legal"
            >
              Legal
            </a>
          </li>
        </ul>
        <div
          class="flex justify-start items-center mt-10 lg:mt-0"
        >
          <a
            class="text-black mr-5"
            data-testid="socials-link"
            href="https://behance.net/adithyabhat"
            rel="noreferrer noopener"
            target="_blank"
          >
            <img
              alt="behance"
              loading="lazy"
              src="/static/behance.svg"
            />
          </a>
          <a
            class="text-black mr-5"
            data-testid="socials-link"
            href="https://dribbble.com/adithyanr"
            rel="noreferrer noopener"
            target="_blank"
          >
            <img
              alt="dribbble"
              loading="lazy"
              src="/static/dribbble.svg"
            />
          </a>
          <a
            class="text-black mr-5"
            data-testid="socials-link"
            href="https://twitter.com/adithya__nr"
            rel="noreferrer noopener"
            target="_blank"
          >
            <img
              alt="twitter"
              loading="lazy"
              src="/static/twitter.svg"
            />
          </a>
          <a
            class="text-black mr-5"
            data-testid="socials-link"
            href="https://github.com/adithyabhat17"
            rel="noreferrer noopener"
            target="_blank"
          >
            <img
              alt="github"
              loading="lazy"
              src="/static/github.svg"
            />
          </a>
          <a
            class="text-black mr-5"
            data-testid="socials-link"
            href="https://linkedin.com/in/adithya-nr"
            rel="noreferrer noopener"
            target="_blank"
          >
            <img
              alt="linkedin"
              loading="lazy"
              src="/static/linkedin.svg"
            />
          </a>
        </div>
      </footer>
    </div>
  </div>
</div>
`;
